File: nodes/json-stringifier/json-stringifier.process.py
import json

async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the JSON Stringifier node.
    Converts an object or array to a JSON string.
    """
    value = inputs.get("value")
    json_string = ""
    error = ""
    success = False
    
    try:
        pretty = settings.get("pretty", False)
        indent = int(settings.get("indent", 2)) if pretty else None
        
        # Set separators to match JavaScript's JSON.stringify output
        # JavaScript doesn't add spaces after colons and commas by default
        if not pretty:
            json_string = json.dumps(value, indent=indent, separators=(',', ':'))
        else:
            json_string = json.dumps(value, indent=indent)
            
        success = True
    except Exception as e:
        error = str(e)
        json_string = "{}"
    
    return {
        "json": json_string,
        "error": error,
        "success": success
    } 

File: nodes/json-stringifier/json-stringifier.process.js
export default async ({inputs, settings, config}) => {

  const value = inputs.value;
  let json = '';
  let error = '';
  let success = false;
  
  try {
    const pretty = settings.pretty === true;
    const indent = pretty ? Number(settings.indent) || 2 : undefined;
    
    json = JSON.stringify(value, null, indent);
    success = true;
  } catch (e) {
    error = e.message || 'Failed to stringify value';
    json = '{}';
  }
  
  return { json, error, success };
}; 